src/Compilers/CSharp/Portable/Syntax/DirectiveTriviaSyntax.cs

```diff
@@ -83,7 +83,7 @@ public SyntaxToken DirectiveNameToken
                             }
                         }
                     }
-                    else if (tr.UnderlyingNode == this.Green)
+                    else if (tr.UnderlyingNode == this.Green && tr.SpanStart == this.SpanStart)
```

Comment:
>tr.UnderlyingNode == this.Green && tr.SpanStart == this.SpanStart [](http://example.com/codeflow?start=29&length=65)

Would something like this work?
```tr.UnderlyingNode == this.Green  && tr.SpanStart == this.SpanStart && tr.IsDirective && (object)tr.GetStructure() == this```

Reply:
More restrictive check sounds good to me. This allowed for a bit of refactoring with the other branches too to simplify things.